simulation/core/create_index.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Create a dataset index file from input parameters."""
+
+import os.path
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+
+import json
+import os
+import random
+import string
+from typing import Optional
+
+
+def get_random_foldername():
+    """Generate random folder name to store the index file in."""
+    return ''.join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits)
+        for _ in range(16))
+
+
+def create_stream_index(shards: int, samples_per_shard: int, avg_raw_shard_size: int,
+                        avg_zip_shard_size: Optional[int]) -> str:
+    """Create dataset index file from input parameters.
+
+    Args:
+        shards (int): Number of shards.
+        samples_per_shard (int): Number of samples per shard.
+        avg_raw_shard_size (int): Average raw shard size.
+        avg_zip_shard_size (int): Average compressed shard size.
+
+    Returns:
+        local path to created index file for stream.
+    """
+    index_data = {'version': 2, 'shards': []}
+
+    shards_list = []
+    for shard_id in range(shards):
+        shard_data = {
+            'column_encodings': [],
+            'column_names': [],
+            'column_sizes': [],
+            'format': 'mds',
+            'raw_data': {
+                'basename': 'shard.' + str(shard_id) + '.mds',
+                'bytes': avg_raw_shard_size,
+                'hashes': {}
+            },
+            'hashes': [],
+            'samples': samples_per_shard,
+            'size_limit': avg_raw_shard_size,
+            'version': 2,
+            'zip_data': None,
+            'compression': None
+        }
+        if avg_zip_shard_size is not None:
+            shard_data['zip_data'] = {
+                'basename': 'shard.' + str(shard_id) + '.mds.zstd',
```

Comment:
Any reason why compression extension is hard coded?

Reply:
Removed all hardcoding from shard basenames in `raw_data` and `zip_data` when creating this index file. All the fields have to be present because otherwise StreamingDataset will throw an error if the index file is not structured correctly, but no hardcoding / making fake shard names is actually needed so I've removed that now.